src/actions/searchActions.ts

```diff
@@ -2,35 +2,51 @@ import { Action, Actions } from "./constants";
 import { ThunkAction } from "redux-thunk";
 
 import { AppState } from "../reducers/store";
-import { MessageI } from "../dataModels/dataModels";
+
+import {
+  _getMessagesAndPoints,
+  populateMessageAndPoints,
+  _PopulateMessageAndPointsParams,
+} from "./dbActions";
 
 export interface SearchByContentParams {
   searchQuery: string;
 }
 
 export interface _SearchByContentParams extends SearchByContentParams {
-  results: MessageI[];
+  results: string[];
 }
 
 export const searchByContent = (
   params: SearchByContentParams
-): ThunkAction<void, AppState, unknown, Action<_SearchByContentParams>> => {
+): ThunkAction<
+  void,
+  AppState,
+  unknown,
+  Action<_SearchByContentParams | _PopulateMessageAndPointsParams>
+> => {
   return (dispatch, getState) => {
-    const state = getState();
+    (async () => {
+      const state = getState();
+
+      if (!state.db.db)
+        return console.warn("Tried to search before database was loaded");
+      const db = state.db.db;
+
+      const pointIds = await db.searchPointsByContent(params.searchQuery);
+      const _results = await db.searchMessagesForPoints(pointIds);
+      const results = _results.map((r) => r._id);
```

Comment:
When ushin-db returns results of type `string[]`, we can rename `_results` to `results` and remove line 38. Easy peasy!